crates/strategies/blend-strategies/src/auction_manager.rs

```diff
@@ -225,6 +226,55 @@ impl OngoingAuction {
         info!("Fill_block: {:?}, profit: {:?} \n bid required: {:?} \n target block: {:?} \n pct to fill: {:?}", fill_block, profit, bid_required, self.target_block, self.pct_to_fill);
         profit
     }
+    pub fn build_requests(
+        &self,
+        wallet: &HashMap<String, i128>,
+        primary_collateral: &String,
+    ) -> Result<Vec<Request>> {
+        let mut requests: Vec<Request> = vec![Request {
+            request_type: 6 + self.auction_type,
+            address: self.user.clone(),
+            amount: self.pct_to_fill as i128,
+        }];
+
+        let mut new_wallet_balance: HashMap<String, i128> = HashMap::new();
```

Comment:
we should clone the existing `wallet_balance` here. Otherwise, the repayments on `ln 270` will pull the wrong value if `auction_type == 1` or if the asset does not exist in the lot.